crates/matrix-sdk/src/client/mod.rs

```diff
@@ -1486,28 +1490,35 @@ impl Client {
         self.inner.http_client.send(request, config, self.server_versions().await?).await
     }
 
-    async fn server_versions(&self) -> HttpResult<Arc<[MatrixVersion]>> {
-        let mut server_versions = self.inner.server_versions.lock().await;
+    async fn request_server_versions(&self) -> HttpResult<Arc<[MatrixVersion]>> {
```

Comment:
You can inline this as an `async {}` block before the two `let server_versions` in the `server_versions` function.

Reply:
Other than that and CI still failing, this looks good now!